Build failure: scummvm

### Steps To Reproduce

Steps to reproduce the behavior:
1. build `scummvm`

### Build log

```
error: builder for '/nix/store/gqvhfmy7d1rvh4ypzshh70i8h6q2qkk5-scummvm-2.8.1.drv' failed with exit code 2;
       last 25 log lines:
       >     C++      engines/ags/plugins/ags_wadjet_util/ags_wadjet_util.o
       >     C++      engines/ags/plugins/ags_waves/ags_waves.o
       >     C++      engines/ags/plugins/ags_waves/data.o
       >     C++      engines/ags/plugins/ags_waves/draw.o
       >     C++      engines/ags/plugins/ags_waves/sound.o
       >     C++      engines/ags/plugins/ags_waves/warper.o
       >     C++      engines/ags/plugins/ags_waves/weather.o
       >     C++      engines/ags/lib/freetype-2.1.3/autohint/ahangles.o
       >     C++      engines/ags/lib/freetype-2.1.3/autohint/ahglobal.o
       >     C++      engines/ags/lib/freetype-2.1.3/autohint/ahglyph.o
       >     C++      engines/ags/lib/freetype-2.1.3/autohint/ahhint.o
       >     C++      engines/ags/lib/freetype-2.1.3/ftgloadr.o
       >     C++      engines/ags/lib/freetype-2.1.3/ftutil.o
       > engines/ags/lib/freetype-2.1.3/autohint/ahglyph.cpp:299:8: error: cannot initialize a variable of type 'char *' with an lvalue of type 'unsigned char *'
       >         char *tag = gloader->current.outline.tags;
       >               ^     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
       > engines/ags/lib/freetype-2.1.3/autohint/ahglyph.cpp:411:10: error: cannot initialize a variable of type 'char *' with an lvalue of type 'unsigned char *'
       >                         char *tag = source->tags;
       >                               ^     ~~~~~~~~~~~~
       > engines/ags/lib/freetype-2.1.3/autohint/ahglyph.cpp:460:11: error: cannot initialize a variable of type 'short *' with an lvalue of type 'unsigned short *'
       >                         short *end = source->contours;
       >                                ^     ~~~~~~~~~~~~~~~~
       > 3 errors generated.
```

### Additional context

N/A

### Notify maintainers

<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->
@peterhoeg 

### Metadata

<!-- Please insert the output of running `nix-shell -p nix-info --run "nix-info -m"` below this line -->
 - system: `"aarch64-darwin"`
 - host os: `Darwin 22.6.0, macOS 13.6.9`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.21.1`
 - channels(root): `"nixpkgs"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs`

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/games/scummvm/default.nix
+++ b/pkgs/games/scummvm/default.nix
@@ -1,4 +1,4 @@
-{ lib, stdenv, fetchFromGitHub, nasm
+{ lib, stdenv, fetchFromGitHub, fetchpatch, nasm
 , alsa-lib, curl, flac, fluidsynth, freetype, libjpeg, libmad, libmpeg2, libogg, libtheora, libvorbis, libGLU, libGL, SDL2, zlib
 , Cocoa, AudioToolbox, Carbon, CoreMIDI, AudioUnit, cctools
 }:
@@ -13,6 +13,14 @@ stdenv.mkDerivation rec {
     rev = "v${version}";
     hash = "sha256-8/q16MwHhbbmUxiwJOHkjNxrnBB4grMa7qw/n3KLvRc=";
   };
+
+  patches = [
+    # Fix building with Freetype 2.13.3. Remove after next release.
+    (fetchpatch {
+      url = "https://github.com/scummvm/scummvm/commit/65977961b20ba97b1213b5267da0cb1efb49063b.patch?full_index=1";
+      hash = "sha256-e5dJd3gP8OAD3hEJlvOhMemsNErCKTn7avlprApFig0=";
+    })
+  ];
 
   nativeBuildInputs = [ nasm ];
 

